crates/host/src/wasmbus/mod.rs

```diff
@@ -450,6 +450,108 @@ impl Bus for Handler {
     }
 }
 
+#[async_trait]
+impl KeyValueReadWrite for Handler {
+    #[instrument]
+    async fn get(
+        &self,
+        bucket: &str,
+        key: String,
+    ) -> anyhow::Result<(Box<dyn AsyncRead + Sync + Send + Unpin>, u64)> {
+        #[derive(Deserialize)]
+        struct GetResponse {
+            #[serde(default)]
+            value: String,
+            #[serde(default)]
+            exists: bool,
+        }
+
+        const METHOD: &str = "wasmcloud:keyvalue/KeyValue.Get";
```

Comment:
Just a heads up that currently the methods are not prefaced with the contract name as it is part of the linkdef. So this should just be `KeyValue.Get`

Reply:
Methods may not contain the package specification in current (legacy) wasmbus invocation scheme, but that's not relevant here, implementation requires the "fully qualified" method name, which must include the package specification by definition.

Currently, links are established per-package, but that will have to change to per-interface links for component model support

Reply:
Ok that makes sense. We should just make sure to update @vados-cosmonic's codegen work before it lands to make sure it does handle the fully qualified messages because I know my hand rolled coded didn't do that. Also, did we capture that change in wasmbus in an issue or quick RFC? Want to make sure that is documented as an implementation detail in our docs page eventually

Reply:
There's no change in wasmbus protocol yet, this is all internal to the host.

_eventually_ we'll have to make changes, I was planning to file an issue today

Reply:
https://github.com/wasmCloud/wasmCloud/issues/458